### PR TITLE
Python 3: port xhr tests [part 7]

### PR DESCRIPTION
**Pull request by ziransun**

### CODE REVIEW

File: xhr/resources/upload.py [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+
 def main(request, response):
     content = []
 
-    for key, values in sorted(item for item in request.POST.items() if not hasattr(item[1][0], "filename")):
-        content.append("%s=%s," % (key, values[0]))
-    content.append("\n")
+    for key, values in sorted(item for item in request.POST.items() if not hasattr(item[1][0], u"filename")):
+        content.append(u"%s=%s," % (isomorphic_decode(key), str(values[0])))
+    content.append(u"\n")
```
Comment (Hexcles) [inline xhr/resources/upload.py:8]: Would it be easier/cleaner to make `content` a bytes string?

**Review by Hexcles — COMMENTED** [review]
Only one question

File: xhr/resources/upload.py (hunk 1 above)
Comment (ziransun) [reply xhr/resources/upload.py]: Code updated. Thank you!

File: xhr/resources/upload.py [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
-    for key, values in sorted(item for item in request.POST.items() if hasattr(item[1][0], "filename")):
+    for key, values in sorted(item for item in request.POST.items() if hasattr(item[1][0], u"filename")):
         value = values[0]
-        content.append("%s=%s:%s:%s," % (key,
-                                         value.filename,
-                                         value.headers["Content-Type"],
-                                         len(value.file.read())))
+        content.append(b"%s=%s:%s:%d," % (key,
+                                          isomorphic_encode(value.filename),
+                                          isomorphic_encode(value.headers[u"Content-Type"]) if value.headers[u"Content-Type"] != None else None,
```
Comment (Hexcles) [inline xhr/resources/upload.py:14]:
```suggestion
                                          isomorphic_encode(value.headers[u"Content-Type"]) if value.headers[u"Content-Type"] is not None else b"None",
```

**Review by Hexcles — APPROVED** [review]
LGTM % nit

**Pull request merged**